packages/react/src/jsx.tsx

```diff
@@ -0,0 +1,51 @@
+import { ComponentProps, ReactElement } from 'react'
+
+import { handler } from './events'
+import { ComponentDataState, JSONValue } from './template'
+
+type Props = Record<string, JSONValue | ((args: any) => any)> & {
+  children: JSONValue
+}
+
+// tbd: investigate this
+export const jsxs = jsx
+
+export function jsx(
+  componentFactory: (props: any) => React.ReactElement<Props>,
+  { children, ...passedProps }: Props,
+  key: string
+): ComponentDataState {
+  const { type, props } = componentFactory(passedProps)
+  if (typeof type !== 'string') {
+    throw new Error('Invalid component.')
+  }
+  const serialisedProps = Object.fromEntries(
+    Object.entries(props).map(([key, value]) => {
+      if (typeof value === 'function') {
+        // tbd: how do we now if it's async or not?
```

Comment:
I wonder, do we need to know if it is async? There should be a way to handle it transparently

On the client we will always have an async function, but on the server it shouldn't matter.. right?

Reply:
So this comment means that if it's async handler, we should wrap it with `asyncHandler`. This registers promise on the client with id, that will be resolved when server sends response back to the client.

In the other PR, we discussed that we would go with async-only behavior, and I mentioned it would simplify stuff once JSX support is in. 

This is the place that would be simplified. We just treat all as async anyway.